hack/build-carvel-packages.sh

```diff
@@ -0,0 +1,435 @@
+#
+set -euo pipefail
+
+#
+# Helper functions
+#
+function log_note() {
+  GREEN='\033[0;32m'
+  NC='\033[0m'
+  if [[ ${COLORTERM:-unknown} =~ ^(truecolor|24bit)$ ]]; then
+    echo -e "${GREEN}$*${NC}"
+  else
+    echo "$*"
+  fi
+}
+
+function log_error() {
+  RED='\033[0;31m'
+  NC='\033[0m'
+  if [[ ${COLORTERM:-unknown} =~ ^(truecolor|24bit)$ ]]; then
+    echo -e "🙁${RED} Error: $* ${NC}"
+  else
+    echo ":( Error: $*"
+  fi
+}
+
+function check_dependency() {
+  if ! command -v "$1" >/dev/null; then
+    log_error "Missing dependency..."
+    log_error "$2"
+    exit 1
+  fi
+}
+
+
+pinniped_path="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+cd "$pinniped_path" || exit 1
+
+# arguments provided to scripts called by hack/prepare-for-integration-tests.sh
+# - app: unimportant, but always first
+# - tag: uuidgen in hack/prepare-for-integration-tests.sh
+#        if this script is run standalone, then auto-fill with a unique value
+app=${1:-"undefined"}
+tag=${2:-$(uuidgen)}
+
+if [[ "${PINNIPED_USE_LOCAL_KIND_REGISTRY:-}" == "" ]]; then
+  log_error "Building the Carvel package requires configuring kind with a local registry."
+  log_error "please set the environment variable PINNIPED_USE_LOCAL_KIND_REGISTRY"
+  log_error "for example:"
+  log_error "    PINNIPED_USE_LOCAL_KIND_REGISTRY=1 ./hack/prepare-for-integration-tests.sh --clean --alternate-deploy ./hack/noop.sh --post-install ./hack/build-carvel-packages.sh"
+fi
+
+
+# TODO: automate the version by release somehow.
+# the tag is the version in our build scripts, but we will want real versions for releases
+pinniped_package_version="${tag}" # ie, "0.25.0"
+
+# core pinniped binaries (concierge, supervisor, local-user-authenticator)
+# TODO: we can likely just pass in the whole registry_repo_tag from the parent script and be done.
+#    the duplication is unnecessary.  This script doesn't ever need to run standalone again.
+registry="kind-registry.local:5000"
+repo="test/build"
+registry_repo="$registry/$repo"
+registry_repo_tag="${registry_repo}:${tag}"
+
+api_group_suffix="pinniped.dev"
+
+# Package prefix for concierge, supervisor, local-user-authenticator
+package_repo_prefix="${registry_repo}/package" # + $resource_name + ":" + $tag
+
+# Pinniped Package repository
+package_repository_repo="pinniped-package-repository"
+package_repository_repo_tag="${registry_repo}/${package_repository_repo}:${tag}"
+
+# carvel
+log_note "Installing kapp-controller on cluster..."
+KAPP_CONTROLLER_GLOBAL_NAMESPACE="kapp-controller-packaging-global"
+kapp deploy --app kapp-controller --file "https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml" -y
+kubectl get customresourcedefinitions
+
+# Generate the OpenAPI v3 Schema files, imgpkg images.yml files
+declare -a arr=("local-user-authenticator" "concierge" "supervisor")
+for resource_name in "${arr[@]}"
```

Comment:
Do we need to keep this loop?
Or could we switch to using just the `app=${1:-"undefined"}` and running it repeatedly?